File: app/src/main/java/com/lumiyaviewer/lumiya/render/spatial/DrawListEntry.java
package com.lumiyaviewer.lumiya.render.spatial;

import com.lumiyaviewer.lumiya.utils.InlineList;
import com.lumiyaviewer.lumiya.utils.InlineListEntry;
import javax.annotation.Nonnull;

public abstract class DrawListEntry implements InlineListEntry<DrawListEntry> {

    @Nonnull
    final float[] boundingBox = new float[6];
    private volatile InlineList<DrawListEntry> list;
    private DrawListEntry next;
    private DrawListEntry prev;

    public abstract void addToDrawList(@Nonnull DrawList drawList);

    @Override
    public InlineList<DrawListEntry> getList() {
        return this.list;
    }

    /* JADX WARN: Can't rename method to resolve collision */
    @Override
    public DrawListEntry getNext() {
        return this.next;
    }

    /* JADX WARN: Can't rename method to resolve collision */
    @Override
    public DrawListEntry getPrev() {
        return this.prev;
    }

    @Override
    public void requestEntryRemoval() {
        InlineList<DrawListEntry> inlineList = this.list;
        if (inlineList != null) {
            inlineList.requestEntryRemoval(this);
        }
    }

    @Override
    public void setList(InlineList<DrawListEntry> inlineList) {
        this.list = inlineList;
    }

    @Override
    public void setNext(DrawListEntry drawListEntry) {
        this.next = drawListEntry;
    }

    @Override
    public void setPrev(DrawListEntry drawListEntry) {
        this.prev = drawListEntry;
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/render/spatial/DrawListEntry.kt
package com.lumiyaviewer.lumiya.render.spatial

import com.lumiyaviewer.lumiya.utils.InlineList
import com.lumiyaviewer.lumiya.utils.InlineListEntry

abstract class DrawListEntry : InlineListEntry<DrawListEntry> {
    @JvmField
    val boundingBox = FloatArray(6)

    @Volatile
    private var list: InlineList<DrawListEntry>? = null
    private var next: DrawListEntry? = null
    private var prev: DrawListEntry? = null

    abstract fun addToDrawList(drawList: DrawList)

    override fun getList(): InlineList<DrawListEntry>? = list

    override fun getNext(): DrawListEntry? = next

    override fun getPrev(): DrawListEntry? = prev

    override fun requestEntryRemoval() {
        list?.requestEntryRemoval(this)
    }

    override fun setList(list: InlineList<DrawListEntry>?) {
        this.list = list
    }

    override fun setNext(next: DrawListEntry?) {
        this.next = next
    }

    override fun setPrev(previous: DrawListEntry?) {
        this.prev = previous
    }
}
